### ISSUE TITLE
[a11y-axe-core]: REACT critical issues

### ISSUE THREAD

**Issue by ariellalgilmore** [carbon-design-system/carbon-for-ibm-dotcom#8353]
### Description

When removing `true` from  the `checkAxeA11y`, I expect everything to pass: https://github.com/carbon-design-system/carbon-for-ibm-dotcom/blob/main/packages/react/tests/e2e-storybook/cypress/support/commands.js#L105

Instead, seeing 5 components fail
<img width="854" alt="Screen Shot 2022-02-22 at 10 05 38 AM" src="https://user-images.githubusercontent.com/20210594/155182363-9b267f64-d18e-4470-b7af-813696dd47e9.png">

Once all pass change `includeImpacts` to include `serious` and create a new issue with `serious` bugs

### Component(s) impacted

Masthead (default, custom, L1, platform)
ToC

### Carbon for IBM.com version

canary

### Severity

Severity 2 = Aspects of design is broken, and impedes users in a significant way, but there is a way to complete their tasks. Affects major functionality, has a workaround.

### Application/website

carbon for ibm.com

### Package

@carbon/ibmdotcom-react

### CodeSandbox example

n/a

### Steps to reproduce the issue (if applicable)

1. `packages/react`
2. change `checkAxeA11y` to remove `true`
3. run `yarn test:e2e-storybook:test:no-percy`

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/blob/main/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/issues) for duplicate issues
